core/app/models/spree/order.rb

```diff
@@ -694,6 +694,10 @@ def covered_by_store_credit?
     end
     alias_method :covered_by_store_credit, :covered_by_store_credit?
 
+    def covered_by_valid_payments?
+      return payments.where(state: %w(checkout pending)).sum(:amount) >= total
```

Comment:
> It would return voided payments as well

Eesh, that's something I didn't realize until just now. I wonder if we really want "voided" payments to be part of the "valid" scope. @jhawthorn @plongyear do either of you know of any code that expects/depends on "voided" payments being "valid"?